grails-app/domain/org/pih/warehouse/shipping/Shipment.groovy

```diff
@@ -196,12 +196,12 @@ class Shipment implements Comparable, Serializable, Historizable {
             })?.unique({ a, b -> a <=> b })?.size() == referenceNumbers?.size()
         })
 
-        // a shipment can't have two events with the same event code (this should be the case for the current event codes: CREATED, SHIPPED, RECEIVED)
-        // we may want to change this in the future?
+        // a shipment can't have two system events with the same event code
```

Comment:
actually what about partially_received events?

Reply:
For now I guess it doesn't matter, because we build history items for partially_received events looking at the `receipts`, not the `events